mantidproject/mantid
# Duplicate signal connections created query loop in Reflectometry Polref GUI

In the `setTableList` method (QtReflMainView.cpp) each entry in the table list had a connection between the entry being chosen and the `setModel(QString)` method.

The `setModel` method then told the Presenter to open the table from the list and the user was them prompted about unsaved changes. If there were more than 1 item in the table list then each table in the list would have an amount of connections equal to the number of list items as they were not being disconnected.

This PR disconnects those previous signals before reconnecting them again.

**To Test:** 
Before testing make sure you have the data from SampleData-ISIS which can be downloaded from [here](http://sourceforge.net/projects/mantid/files/Sample%20Data/SampleData-ISIS.zip/download)
- Manage your data directories to point to the location of INTER_NR.tbl found inside SampleData-ISIS
- Go to Interfaces -> Reflectometry -> ISIS Reflectometry (Polref)
- Click Reflectometry on the menu bar of the interface
- Click Import .TBL
- Load the INTER_NR.tbl file from SampleData-ISIS and give it a workspace name
- Load the same table as many times as you like, giving it a different workspace name each time
- Edit the table currently in the processing view
- Go to `Reflectometry` on the menu bar and then `Open Table`
- Choose another table to load in
- A prompt will appear telling you that you have unsaved changes, would you like to discard them?
- Press `No`
- Ensure that the pop up does not reappear
- Repeat the edit and open another table step a few times to make sure this behaviour remains constant

**Release Notes:** http://www.mantidproject.org/Release_Notes_3_6_Reflectometry

Fixes #14947
